Mathlib/LinearAlgebra/TensorProduct/Subalgebra.lean

```diff
@@ -128,11 +127,83 @@ theorem comm_trans_rTensorBot :
     (Algebra.TensorProduct.comm R _ _).trans A.rTensorBot = A.lTensorBot :=
   AlgEquiv.toLinearEquiv_injective (toSubmodule A).comm_trans_rTensorOne
 
+variable (R S T)
+
+/-- Given `R`-algebras `S,T`, there is a natural `R`-linear isomorphism from `S ⊗[R] T` to
+`S' ⊗[R] T'` where `S',T'` are images of `S,T` in `S ⊗[R] T`.
+This is promoted to an `R`-algebra isomorphism `Algebra.TensorProduct.algEquivIncludeRange`. -/
+def _root_.Algebra.TensorProduct.linearEquivIncludeRange :
```

Comment:
Is this the best place for these lemmas? Can you put them at a place where you can close the current namespace and start
```
namespace Algebra.TensorProduct
```

Reply:
I tried to move this definition to an earlier file but seems that it's not possible; it uses `Submodule.mulMap` which is defined in the file `LinearAlgebra/TensorProduct/Submodule` which is just imported by this file, and it uses `Algebra.TensorProduct.*` which is not imported by `LinearAlgebra/TensorProduct/Submodule`.

Let me rearrange the namespaces a little bit.

Reply:
You might try to add imports to earlier files as long as not too many.

Reply:
I'm not saying you should move them to a different file. But maybe a slightly different place in this file.

Reply:
> You might try to add imports to earlier files as long as not too many.

Ah, I intentionally not import `Algebra.TensorProduct.*` in the file `LinearAlgebra/TensorProduct/Submodule`. Otherwise the whole contents in this file should be moved to that file.

> I'm not saying you should move them to a different file. But maybe a slightly different place in this file.

Done.